scripts/build-docker-image.sh

```diff
@@ -0,0 +1,70 @@
+#! /bin/sh
+
+root=${PWD}
+repo_dir="$root"/demo-repository
+image_name="learnocaml-docker"
+
+function print_usage() {
+    printf "Usage: %s <OPTIONS>\n\
+Options:\n\
+  -repo-dir <directory> (default = demo-repository): \n\
+\t Repository containing the exercises.\n\
+  -root <directory> (default = %s): \n\
+\t Root of the learn-ocaml repository.\n" "$0" "$pwd"
+}
+
+function option_error () {
+    printf "Incorrect option $1\n" 1>&2
+    print_usage
+    exit 2
+}
+
+function option_value_error () {
+    printf "Incorrect value $2 for option $1\n" 1>&2
+    print_usage
+    exit 2
+}
+
+function check_directory () {
+    if ! [ -d "$2" ]; then
+        printf "$2 is not a valid directory.\n" 1>&2
+        option_value_error "$1" "$2"
+    fi
+}
+
+while [[ $# -gt 0 ]]; do
+  curr="$1"
+
+  case $curr in
+      -image-name)
+          image_name="$2"
+          shift 2
+          ;;
+      -repo-dir)
+          check_directory "$1" "$2"
+          repo_dir="$2"
+          shift 2
+          ;;
+      -root)
+          check_directory "$1" "$2"
+          root="$2"
+          shift 2
+          ;;
+      *)    # unknown option
+          option_error $1
+          ;;
+  esac
+done
+
+exercises_repository="$root"/exercises_repository
+      
+if [ -d "$execises_repository" ]; then
```

Comment:
According to my remark on `print_usage`, `"$root"/` should be removed. Right now, it doesn't even accept an absolute path.

Reply:
I deleted my previous comment, since it was related to my previous one on the `-root` option.

I don't think there is a typo here. Before building the docker image, we must copy the exercises into a local temporary directory, which is called `exercises_repository`. I suppose `shellcheck` thinks that line 59, the second `exercises_repository` is a reference to the variable of the same name.

Reply:
`shellcheck` just said the variable is undefined as `execises` should be `exercises` :)

Reply:
Oh, I should've read more carrefully! Thanks. (I'm too much used to OCaml's static typing)